### PR TITLE
Take into account branches in case of stack max size calculation

### PR DESCRIPTION
**Pull request by jakubch1**
Fix for #443 

Few questions based on comment in code:
https://github.com/microsoft/CLRInstrumentationEngine/blob/main/src/InstrumentationEngine/InstructionGraph.cpp#L1094

1) should I remove this comment?
2) should I add similar logic for switches?
3) do you have any framework for testing this max stack size?

### CODE REVIEW

**Comment by jakubch1** [comment]
@WilliamXieMSFT @wiktork  I've tested changes on roslyn and repo which discovered the initial issue and changes looks good. Not getting any InvalidProgramExceptions. Please review changes.

File: src/InstrumentationEngine/InstructionGraph.cpp [hunk 1]
```diff
@@ -1091,28 +1091,89 @@ HRESULT MicrosoftInstrumentationEngine::CInstructionGraph::CalculateMaxStack(_Ou
...
+                it->second = stackDepth;
+            }
+        }
+        else
+        {
+            stackDepthMap.insert(make_pair(offset, stackDepth));
+        }
+    };
+
+    // Heuristic to calculate maxstack. This may guess sligthly high.
```
Comment (wiktork) [inline src/InstrumentationEngine/InstructionGraph.cpp:1112]: nit: slightly

File: tests/InstrEngineTests/InstrEngineTests/EmbeddedResources/MaxStackSizeExample.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace MaxStackSizeExample
+{
+    class Program
+    {
+        static int Main(string[] args)
+        {
+            GetHdiPolicyValidationRequest(new HdiJobMetadata()
```
Comment (wiktork) [inline tests/InstrEngineTests/InstrEngineTests/EmbeddedResources/MaxStackSizeExample.cs:13]: Are these type names from internal code? Did we want to rename them?

**Review by wiktork — APPROVED** [review]

File: src/InstrumentationEngine/InstructionGraph.cpp (hunk 1 above)
Comment (jakubch1) [reply src/InstrumentationEngine/InstructionGraph.cpp]: fixed

File: src/InstrumentationEngine/InstructionGraph.cpp [hunk 3]
```diff
@@ -1149,14 +1210,26 @@ HRESULT MicrosoftInstrumentationEngine::CInstructionGraph::IsFirstInstructionInC
         DWORD exceptionFlags;
         IfFailRet(pExceptionClause->GetFlags(&exceptionFlags));
 
-        if (exceptionFlags == COR_ILEXCEPTION_CLAUSE_NONE)
+        if (exceptionFlags == COR_ILEXCEPTION_CLAUSE_NONE || exceptionFlags == COR_ILEXCEPTION_CLAUSE_FILTER)
```
[the remark this answers is not in the document]
Comment (jakubch1) [reply src/InstrumentationEngine/InstructionGraph.cpp:1213]: Filter contains both filter and handler. First if is changing stack size to 1 for handler and second if for filter. So this is needed in both places

**Pull request merged**